### PR TITLE
Accept 202 http status code (instead throwing exception). 

### PR DESCRIPTION
**Pull request by robocik**
This status is used by WCF IsOneWay operations

### CODE REVIEW

**Comment by Kisty** [comment]
Can't you just wrap it in a try catch block and not throw exception if the status is 202. Flatly accepting it for all users of the library is a little dangerous.

The status code is accepted but doesn't verify if the process has been processed (which would be 200 OK). Normally we should expect 200 OK every time. I think it's fair to throw an exception.

From W3 HTTP Standards

> ## 10.2.3 202 Accepted
> 
> The request has been accepted for processing, but the processing has not been completed. The request might or might not eventually be acted upon, as it might be disallowed when processing actually takes place. There is no facility for re-sending a status code from an asynchronous operation such as this.
> 
> The 202 response is intentionally non-committal. Its purpose is to allow a server to accept a request for some other process (perhaps a batch-oriented process that is only run once per day) without requiring that the user agent's connection to the server persist until the process is completed. The entity returned with this response SHOULD include an indication of the request's current status and either a pointer to a status monitor or some estimate of when the user can expect the request to be fulfilled.

**Comment by Kisty** [comment]
Actually, checking the OkHttp3 library, a response is counted as a success if the response is in between 200 and 300. Do you think you could change the status check to that?

https://github.com/square/okhttp/blob/7135628c645892faf1a48a8cff464e0ed4ad88cb/okhttp/src/main/java/okhttp3/Response.java#L105

File: ksoap2-j2se/src/main/java/org/ksoap2/transport/HttpTransportSE.java [hunk 1]
```diff
@@ -271,7 +271,10 @@ public List call(String soapAction, SoapEnvelope envelope, List headers, File ou
             is = readDebug(is, contentLength, outputFile);
         }
 
-        parseResponse(envelope, is,retHeaders);
+        if(is!=null)
+        {
+            parseResponse(envelope, is,retHeaders);
+        }
```
Comment (Kisty) [inline ksoap2-j2se/src/main/java/org/ksoap2/transport/HttpTransportSE.java]: Is the null check needed here?

**Comment by robocik** [comment]
To change success condition to code >= 200 && code < 300 we need to make a bigger investigation.  So as a safer solution I would merge the current pull request and later if we ensure that we can use your condition, then we will create a new pull request.

We have to check for null becaues during my test when WCF service returns from IsOneWay method, then response length was 0 and the response stream was null. Which of course throws exception in parseResponse method.

**Comment by Kisty** [comment]
@robocik Ah right. That makes sense. Thanks for explaining :)

**Comment by mosabua** [comment]
Could add some changelog entry and if necessary some documentation?

**Comment by Kisty** [comment]
@mosabua What changes would we like to be applied? The ability to accept status code 202 as well or any code which is `200 <= code && code < 300`? Is there anything else we want added?

**Comment by mosabua** [comment]
I can do the rest tonight if this is good to go from your perspective... @robocik @Kisty

**Comment by Kisty** [comment]
If you mean adding the broader status check and adding documentation surrounding that then yes, that's fine with me.

**Pull request merged**